File: LotDspWeb/src/ngyshanai/web/service/kaigiyoyaku/KaigiYoyakuService.java
package ngyshanai.web.service.kaigiyoyaku;

import java.util.Date;
import java.util.List;

import javax.ejb.EJB;
import javax.ws.rs.DELETE;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;

import ngyshanai.application.LogicExecutor;
import ngyshanai.common.msg.kaigiyoyaku.KaigishitsuInfo;
import ngyshanai.common.msg.kaigiyoyaku.YoyakuInfo;
import ngyshanai.domain.logic.kaigiyoyaku.KaigiYoyakuLogic;
import ngyshanai.domain.logic.kaigiyoyaku.KaigiYoyakuUpdateLogic;
import ngyshanai.domain.logic.kaigiyoyaku.MaishuYoyakuDeleteLogic;
import ngyshanai.domain.logic.kaigiyoyaku.MaishuYoyakuLogic;
import ngyshanai.domain.logic.kaigiyoyaku.SearchKaigishitsuLogic;
import ngyshanai.domain.logic.kaigiyoyaku.YoyakuDeleteLogic;

@Path("/yoyaku")
public class KaigiYoyakuService {

	@EJB
	private LogicExecutor logicExecutor;
	
	@GET
	@Path("/searchkaigishitsu/{hizuke}")
	public List<KaigishitsuInfo> searchkaigishitsu(@PathParam("hizuke") Date hizuke) {
		return logicExecutor.execute(SearchKaigishitsuLogic.class, hizuke);
	}
	
	@POST
	@Path("/save")
	public boolean save(YoyakuInfo info) {
		return logicExecutor.execute(KaigiYoyakuLogic.class, info);
	}
	
	@POST
	@Path("/update")
	public boolean update(YoyakuInfo info) {
		return logicExecutor.execute(KaigiYoyakuUpdateLogic.class, info);
	}
	
	@POST
	@Path("/{maishuEnd}")
	public boolean maishusave(@PathParam("maishuEnd")Date maishuEnd, YoyakuInfo info) {
		return logicExecutor.execute(MaishuYoyakuLogic.class, maishuEnd, info);
	}
	
	@DELETE
	@Path("/{yoyakuCd}")  
	public boolean delete(@PathParam("yoyakuCd") String yoyakuCd) {
		return logicExecutor.execute(YoyakuDeleteLogic.class, yoyakuCd);
	}
	
	@DELETE
	@Path("/{kaigishitsuCd}/{yoyakuDate}/{yoyakuBlockStart}/{maishuYoyakuCd}")  
	public boolean maishudelete(@PathParam("kaigishitsuCd") String kaigishitsuCd, @PathParam("yoyakuDate") Date yoyakuDate, @PathParam("yoyakuBlockStart") String yoyakuBlockStart, @PathParam("maishuYoyakuCd")String maishuYoyakuCd) {
		return logicExecutor.execute(MaishuYoyakuDeleteLogic.class, kaigishitsuCd, yoyakuDate, yoyakuBlockStart, maishuYoyakuCd);
	}
	
}

File: LotDspWeb/WebContent/app/main/kaigishitsuyoyaku/KaigiYoyakuService.js
/**
 * _サービス
 */
(function () {

    // 必要な依存を列挙
    // 列挙の順番（カテゴリー毎に改行を）
    // 1.angular系
    // 2.外部プラグイン系
    // 3.NIS開発系のうち、共通のもの
    // 4.NIS開発系のうち、固有のもの
    // 5.resolveで渡される値
    var injectParams = ['$http', 'NIS'];

    // 引数は依存の内容と一致する
    var newService = function ($http, NIS) {
        
    	let baseURI = 'service/yoyaku';
    	
    	return {
    		search(hizuke) {
    			return NIS.u.req($http.get(NIS.u.path(baseURI, 'searchkaigishitsu', hizuke)));
    		},
    		save(yoyakuInfo) {
    			return NIS.u.req($http.post(NIS.u.path(baseURI, 'save'), yoyakuInfo));
    		},
    		update(yoyakuInfo) {
    			return NIS.u.req($http.post(NIS.u.path(baseURI, 'update'), yoyakuInfo));
    		},
    		maishusave(yoyakuInfo, maishuEnd) {
    			return NIS.u.req($http.post(NIS.u.path(baseURI, maishuEnd), yoyakuInfo));
    		},
    		yoyakudelete(yoyakuCd) {
    			return NIS.u.req($http.delete(NIS.u.path(baseURI, yoyakuCd), {headers: {
                    'Content-Type': 'application/json'
                }}));
    		},
    		maishuyoyakudelete(kaigishitsuCd, yoyakuDate, yoyakuBlockStart, maishuYoyakuCd) {
    			return NIS.u.req($http.delete(NIS.u.path(baseURI, kaigishitsuCd, yoyakuDate, yoyakuBlockStart, maishuYoyakuCd), {headers: {
                    'Content-Type': 'application/json'
                }}));
    		},
    	}
    };

    newService.$inject = injectParams;

    angular.module('app')
        // factory | service | provider
        .factory('KaigiYoyakuService', newService);

}());